src/main/java/uk/ac/ebi/atlas/search/GeneSearchDao.java

```diff
@@ -134,11 +139,14 @@ public GeneSearchDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate,
     // taking about 50 seconds. See https://www.pivotaltracker.com/story/show/173033902 for a full report and detailed
     // benchmarks.
     private static final String SELECT_PREFERRED_K_AND_MIN_P_CLUSTER_ID_FOR_GENE_STATEMENT =
-            "SELECT k, cluster_id FROM scxa_marker_genes AS markers " +
-                    "WHERE experiment_accession=:experiment_accession " +
-                        "AND gene_id=:gene_id " +
-                        "AND marker_probability<:threshold " +
-                        "AND (k=:preferred_K OR marker_probability=:min_marker_probability)";
+			"SELECT variable as k, value as cluster_id " +
+					"FROM scxa_cell_group AS cell_group " +
+					"INNER JOIN scxa_cell_group_marker_genes AS marker_genes " +
+					"ON marker_genes.marker_probability < :threshold " +
+					"AND (variable = :preferred_K OR marker_probability = :min_marker_probability)" +
+					"AND cell_group.id = marker_genes.cell_group_id AND gene_id = :gene_id " +
+					"WHERE experiment_accession = :experiment_accession;";
```

Comment:
Better to keep the ON clause clean in the JOIN to show how you are putting the tables together and leave the value selections for the WHERE clause.

```suggestion
					"ON cell_group.id = marker_genes.cell_group_id "+ 
					"WHERE marker_genes.marker_probability < :threshold " +
					"AND (variable = :preferred_K OR marker_probability = :min_marker_probability)" +
					"AND gene_id = :gene_id " +
					"AND experiment_accession = :experiment_accession;";
```